lib/LedgerSMB/Scripts/payment.pm

```diff
@@ -1677,178 +1673,121 @@ sub use_overpayment2 {
     }
 
 
-    $count=1;
     #lets see which invoice do we have printed
-    while ($Payment->{"entity_id_$count"})
+    for my $count(1 .. $request->{count})
     {
-        if ($Payment->{"checkbox_$count"})
+        # we should rename the checkbox to remove_ or ignore_
+        if ($Payment->{"checkbox_$count"} or not $Payment->{"entity_id_$count"})
         {
             $count++;
             next;
         }
 
-        if ($ovp_repeated_invoices{qq|$Payment->{"invoice_id_$count"}|}->{qq|$Payment->{"selected_accno_$count"}|}
-            != $Payment->{"selected_accno_$count"}) {
-
-            # the "ovp_repeated_invoices" hash will store the conbination
-            # of invoice id and overpayment account, if this convination has
-            # already printed do not print it again
-            $ovp_repeated_invoices{$Payment->{"invoice_id_$count"}}->{$Payment->{"selected_accno_$count"}} =
-                $Payment->{"selected_accno_$count"};
-
-            # the "repeated invoice" flag will check if this invoice has
-            # already been printed, if it does, do not print the apply
-            # discount checkbox in the UI
-
-            my $ovp_inv_payment =
-                $ovp_repeated_invoices{$Payment->{"invoice_id_$count"}};
-            if (! $ovp_inv_payment->{repeated_invoice}){
-                $ovp_inv_payment->{optional_discount} =
-                    $Payment->{"optional_discount_$count"};
-                $ovp_inv_payment->{repeated_invoice} = 'false';
-            } else{
-                $ovp_inv_payment->{repeated_invoice} = 'true';
-            }
-
-            $ui_to_use_subtotal += $Payment->{"amount_$count"};
+        $ui_to_use_subtotal += $Payment->{"amount_$count"};
 
-            my ($id,$name) = split(/--/, $Payment->{"entity_id_$count"});
-            my ($ovp_chart_id, $ovp_selected_accno) =
+        my ($id,$name) = split(/--/, $Payment->{"entity_id_$count"});
+        my ($ovp_chart_id, $ovp_selected_accno) =
                 split(/--/, $Payment->{"selected_accno_$count"});
-            my $applied_due =
-                ($ovp_inv_payment->{optional_discount})
-                ? $Payment->{"due_$count"}
-            : $Payment->{"due_$count"} + $Payment->{"discount_$count"};
-
-            $amount_to_be_used{"$ovp_selected_accno"} +=
-                $Payment->{"amount_$count"};
-            # this hash will keep track of the amount to be paid of an
-            # specific invoice_id, this amount could not be more than the
-            # due of that invoice.
-            $invoice_id_amount_to_pay{qq|$Payment->{"invoice_id_$count"}|} +=
-                $Payment->{"amount_$count"};
-            if($invoice_id_amount_to_pay{qq|$Payment->{"invoice_id_$count"}|}
-               > $applied_due) {
-                $warning .= $locale->text('The amount of the invoice number').qq| $Payment->{"invnumber_$count"} |.$locale->text('is lesser than the amount to be paid').qq|\n|;
```

Comment:
This use of $locale->text is so fundamentally broken I don't see a problem with introducing a new untranslated string in its place.